### ISSUE TITLE
[auto/go,nodejs,python] Add --config-file option to Automation API Config Set

### ISSUE THREAD

**Issue by zbuchheit** [pulumi/pulumi#17927]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

Currently, in the go automation api, you cannot use the --config-file flag with config. It would be nice to be able to use this.

- [x] Go
- [ ] Nodejs
- [ ] Python
- [ ] .NET

### Affected area/feature

Automation API

**Comment by zbuchheit** [comment]
related to https://github.com/pulumi/pulumi/issues/17773

**Comment by tgummerer** [comment]
Reopening this, as I don't think this is actually done in https://github.com/pulumi/pulumi/pull/19222.  This issue talks about the config commands, e.g. `getAllConfig` in NodeJS.  https://github.com/pulumi/pulumi/pull/19222 only addressed stack operations such as `up`, `preview`, `refresh` and `destroy`.

**Comment by rshade** [comment]
Aren't those the only operations that would take a stack config?

**Comment by tgummerer** [comment]
No, `pulumi config` takes a `--config-file` arg. As I mentioned in my comment, `getAllConfig` in NodeJS at least should have it to address this issue.  This is the PR that did it for Go: https://github.com/pulumi/pulumi/pull/17939
